__tests__/app.test.js

```diff
@@ -244,7 +244,161 @@ describe.only('POST /api/articles/:article_id/comments', () => {
+        const invalidVote = { inc_votes: '' };
+        return request(app)
+            .patch('/api/articles/1')
+            .send(invalidVote)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe('bad request');
+            });
+    });
+    test('404 - responds with an error when the article can not be found and vote can not be updated ', () => {
+        const newVote = { inc_votes: 1 };
+        return request(app)
+            .patch('/api/articles/99')
+            .send(newVote)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe('not found');
+            });
+    });
+    test('204 - responds with no updates in votes and empty body', () => {
+        const newVote = { inc_votes: 0 };
+        return request(app)
+            .patch('/api/articles/9')
+            .send(newVote)
+            .expect(204)
+            .then(({ body }) => {
+                expect(body).toEqual({});
+            });
+    });
+})
+
+describe('/api/comments/:comment_id', () =>{
+    test('DELETE: 204 - deletes the specified comment and responds with empty body ', () => {
+        return request(app)
+            .delete('/api/comments/5')
+            .expect(204)
+    });
+    test('DELETE:400 responds with bad request error message when given an invalid id', () => {
+        return request(app)
+        .delete('/api/comments/not-a-number')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('bad request');
+      });
+    });
+    test('DELETE:404 responds with an error message when given a non-existent id', () => {
+        return request(app)
+          .delete('/api/comments/999')
+          .expect(404)
+          .then(({body}) => {
+            expect(body.msg).toBe('not found');
+          });
+      });
+    test('DELETE:404 responds with an error message when remove a comment that has been deleted previously', () => {
+        return request(app)
+        .delete('/api/comments/5')
+        .expect(204)
+        .then(() => {
+            return request(app)
+            .delete('/api/comments/5')
+            .expect(404)
+            .then(({body}) => {
+            expect(body.msg).toBe('not found');
+          });
+        })
         
+      });
+})
+
+describe('/api/users', () => {
+    test('GET - 200 responds an array of objects which contains all users data', () => {
+        return request(app)
+        .get('/api/users')
+        .expect(200)
+        .then(({body}) =>{
+            expect(body.users).toBeInstanceOf(Array);
+            expect(body.users.length).toBe(4);
+            body.users.forEach((user) => {
+                expect(user).toHaveProperty('username');
+                expect(user).toHaveProperty('name');
+                expect(user).toHaveProperty('avatar_url');
+            })
+         })
     })
+})
 
```

Comment:
Here is would be nice to see 200 tests for the `sort_by` and `order` queries individually, since you want to make this endpoint as flexible as possible